### PR TITLE
TEST: add VCR recording workflow

### PR DESCRIPTION
**Pull request by igor-davidyuk**
This PR introduces a manual GItHub action workflow that rebuilds VCR cassettes and pushes it back to the branch

### CODE REVIEW

File: .github/workflows/record-vcr-cassettes.yaml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  # Artifacts and test reports
+  ARTIFACT_DIRECTORY: artifacts
+  REPORT_DIRECTORY: reports
+
+  # Python environment
+  PYTHON_VERSION: 3.9
+
+permissions:
+  contents: read
+  pull-requests: write  # 'write' access to pull requests
```
[the remark this answers is not in the document]
Comment (igor-davidyuk) [reply .github/workflows/record-vcr-cassettes.yaml:22]: I imagine it works like this:
You create a PR with communication-breaking changes and have the tests fail. Then you run this workflow and it updates the cassettes right in your PR. This is why I have this permission in place

File: .github/workflows/record-vcr-cassettes.yaml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+          pip install ".[dev]"
+
+      - name: Create artifact directories
+        # Creates the temporary directories that are used to store the test
+        # artifacts and reports
+        run: |
+          mkdir $ARTIFACT_DIRECTORY
+          mkdir $REPORT_DIRECTORY
+
+      - name: Nightly tests
```
[the remark this answers is not in the document]
Comment (igor-davidyuk) [reply .github/workflows/record-vcr-cassettes.yaml]: done!

File: .github/workflows/record-vcr-cassettes.yaml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  # Login details for the Geti instance to run the tests against
+  # Secrets are configured on repository level (Settings > Secrets)
+  GETI_HOST: ${{ secrets.GETI_HOST }}
+  GETI_USERNAME: ${{ secrets.GETI_USERNAME }}
+  GETI_PASSWORD: ${{ secrets.GETI_PASSWORD }}
+
+  # Other environment variables used in the tests
+  TEST_MODE: "RECORD"
+  # Artifacts and test reports
+  ARTIFACT_DIRECTORY: artifacts
```
[the remark this answers is not in the document]
Comment (igor-davidyuk) [reply .github/workflows/record-vcr-cassettes.yaml]: removed!

File: .github/workflows/record-vcr-cassettes.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ljcornel) [reply .github/workflows/record-vcr-cassettes.yaml:22]: I see where you're coming from but I'm actually not 100% sure about the exact scope of the permissions. I thought that for pushing to any branch you'd need to have `contents: write` permissions, but if the PR is made from a fork of the repo then you're probably right and you'd need to be able to write to pull-requests as well.

Let's keep it as is and see if it works, we can always update them later.

**Review by ljcornel — APPROVED** [review]

**Pull request merged**